crates/subspace-farmer/src/commitments/databases.rs

```diff
@@ -13,14 +13,44 @@ use tracing::error;
 // Cache size is just enough for last 2 salts to be stored
 const COMMITMENTS_CACHE_SIZE: usize = 2;
 
+#[derive(Deref, DerefMut)]
+pub(super) struct CommitmentDb(pub Arc<Db>);
+
+impl CommitmentDb {
+    pub fn open_or_create(path: PathBuf) -> Result<Self, CommitmentError> {
+        let opts = Options {
+            path,
+            columns: vec![ColumnOptions {
+                preimage: false,
+                btree_index: true,
+                uniform: false,
+                ref_counted: false,
+                compression: CompressionType::NoCompression,
+                compression_threshold: 4096,
+            }],
+            sync_wal: true,
+            sync_data: true,
+            stats: false,
+            salt: None,
+        };
+        Db::open_or_create(&opts)
+            .map(Arc::new)
+            .map(CommitmentDb)
+            .map_err(CommitmentError::CommitmentDb)
+    }
+}
```

Comment:
Honestly I don't quite understand why this wrapper is useful. It doesn't really add anything, rather created another layer of indirection seemingly unnecessarily.

Reply:
Hmm, I just thought that you asked to abstract db creation instead of exposing database options to `commitments.rs`. How would you solve that? :thinking: